### PR TITLE
API: make Timestamp/Timedelta _as_unit public as_unit

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the Github issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/_libs/tslibs/nattype.pyx [hunk 1]
```diff
@@ -1210,6 +1211,22 @@ default 'raise'
...
+        """
+        Convert the underlying int64 representaton to the given unit.
+
+        Parameters
+        ----------
+        unit : {"ns", "us", "ms", "s"}
+        round_ok : bool, default True
+            If False and the conversion requires rounding, raise.
+
+        Returns
```
Comment (mroeschke) [inline pandas/_libs/tslibs/nattype.pyx:1224]: Could you add an `Examples` section too? (Hoping to have all public APIs have examples in the near future)

Comment (jbrockmendel) [reply pandas/_libs/tslibs/nattype.pyx:1224]: that would make sense, but first i think we should determine if we want unit to somehow show up in the repr

Comment (mroeschke) [reply pandas/_libs/tslibs/nattype.pyx:1224]: Hmm, frequency didn't end up in the repr before right? (just as a comparison note)

Comment (jbrockmendel) [reply pandas/_libs/tslibs/nattype.pyx:1224]: In 1.5.x `freq` is in the repr when it is non-None

Comment (mroeschke) [reply pandas/_libs/tslibs/nattype.pyx:1224]: Might be helpful then to have the unit resolution show up in the repr then (guess it would be an API change)

Comment (jbrockmendel) [reply pandas/_libs/tslibs/nattype.pyx:1224]: yah that's on my todo list, just not a priority ATM

**Comment by jbrockmendel** [comment]
@jreback mentioned this earlier this week

**Comment by jbrockmendel** [comment]
Following today's call, I'm leaning towards changing "reso" to "creso" for the cdef methods and "unit" to "reso" for the python methods.

**Comment by mroeschke** [comment]
> I think the public-facing keyword/attribute/method should be unit/unit/as_unit

Yeah I would be okay if the "problematic" situation of `to_datetime/to_timedelta/Timestamp/TImedelta(ints, unit="s")` to interpret `[ints]` as epoch seconds _and_ having the resulting unit be seconds, so having the public naming be `unit` would be okay to me.

**Comment by jbrockmendel** [comment]
all good here?  we have the option of changing unit->reso later, but for now id like to get started on some follow-ups to this.

**Review by mroeschke — APPROVED** [review]
LGTM. I am still partial to `unit` over `reso` (for now). Good to have examples in `as_unit` once the unit-in-the-repr question is resolved.

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**